test/parallel/test-cli-eval.js

```diff
@@ -229,3 +229,48 @@ child.exec(`${nodejs} --use-strict -p process.execArgv`,
     assert.strictEqual(err, null);
   }));
 });
+
+// ESModule eval tests
+
+
+// Assert that "42\n" is written to stdout on module eval.
+const execOptions = '--experimental-modules --input-type module';
+child.exec(
+  `${nodejs} ${execOptions} --eval "console.log(42)"`,
+  common.mustCall((err, stdout) => {
+    assert.ifError(err);
+    assert.strictEqual(stdout, '42\n');
+  }));
+
+// Assert that "42\n" is written to stdout with print option.
+child.exec(
+  `${nodejs} ${execOptions} --print --eval "42"`,
+  common.mustCall((err, stdout) => {
+    assert.ifError(err);
+    assert.strictEqual(stdout, '42\n');
+  }));
+
+// Assert that error is written to stderr on invalid input.
+child.exec(
+  `${nodejs} ${execOptions} --eval "!!!!"`,
+  common.mustCall((err, stdout, stderr) => {
+    assert.ok(err);
+    assert.strictEqual(stdout, '');
+    assert.ok(stderr.indexOf('SyntaxError: Unexpected end of input') > 0);
+  }));
+
+// Assert that require is undefined in ESM support
+child.exec(
+  `${nodejs} ${execOptions} --eval "console.log(typeof require);"`,
+  common.mustCall((err, stdout) => {
+    assert.ifError(err);
+    assert.strictEqual(stdout, 'undefined\n');
+  }));
```

Comment:
We probably don't need this test, but there's nothing wrong with including it.

Reply:
It may be valuable to capture the default behavior. If we'd ever (accidentally) flip the default type, this would fail. Though realistically a lot of tests would fail then...